asserts/hardware_identity_test.go

```diff
@@ -0,0 +1,103 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts_test
+
+import (
+	"strings"
+
+	"github.com/snapcore/snapd/asserts"
+	. "gopkg.in/check.v1"
+)
+
+
+var _ = Suite(&hardwareIdentitySuite{})
+
+type hardwareIdentitySuite struct {}
+
+const errPrefix = "assertion hardware-identity: "
+
+const (
+	hardwareIdentityExample = `type: hardware-identity
+authority-id: account-id-1
+issuer-id: account-id-1
+manufacturer: some-manufacturer
+hardware-name: raspberry-pi-4gb
+sign-key-sha3-384: t9yuKGLyiezBq_PXMJZsGdkTukmL7MgrgqXAlxxiZF4TYryOjZcy48nnjDmEHQDp
+hardware-id: random-id-1
+hardware-id-key: TUlHZk1BMEdDU3FHU0liM0RRRUJBUVVBQTRHTkFEQ0JpUUtCZ1FDaVFjaFlNVFAra25jNnZtUFd3SC9tMThqbApIRVN5U0wyZFBIb25lQ1dnOUFuMlM0N3ZBQ1VJd3ZlU0FDRHFHam5Ld3JvcFM3Rmw1cTVFTTZNQXlIUElkSmJwCmdXUFV6bHJBRTRZc2M4VDh5QTUwUlRPaVFPc0x4MHZUMnFHU0kzYk16bFU3bkhyZW0zWXRNOUErbjlGUEVuOVAKT2hNaGVyZkExekVFVmkvSWZ3SURBUUFC
+hardware-id-key-sha3-384: 20HX1SS8dJW8tNCmybAdP7frkn0dmEV9DdwIABskpSsxBUaylHA6oO8dSj+ORWXa
+
+AXNpZw==`
+)
+
+
+func (s *hardwareIdentitySuite) TestDecodeOK(c *C) {
+	a, err := asserts.Decode([]byte(hardwareIdentityExample))
+	c.Assert(err, IsNil)
+	c.Check(a, NotNil)
+	c.Check(a.Type(), Equals, asserts.HardwareIdentityType)
+
+	req := a.(*asserts.HardwareIdentity)
+	c.Check(req.IssuerID(), Equals, "account-id-1")
+	c.Check(req.Manufacturer(), Equals, "some-manufacturer")
+	c.Check(req.HardwareName(), Equals, "raspberry-pi-4gb")
+	c.Check(req.HardwareID(), Equals, "random-id-1")
+	c.Check(req.HardwareIDKey(), Equals, "TUlHZk1BMEdDU3FHU0liM0RRRUJBUVVBQTRHTkFEQ0JpUUtCZ1FDaVFjaFlNVFAra25jNnZtUFd3SC9tMThqbApIRVN5U0wyZFBIb25lQ1dnOUFuMlM0N3ZBQ1VJd3ZlU0FDRHFHam5Ld3JvcFM3Rmw1cTVFTTZNQXlIUElkSmJwCmdXUFV6bHJBRTRZc2M4VDh5QTUwUlRPaVFPc0x4MHZUMnFHU0kzYk16bFU3bkhyZW0zWXRNOUErbjlGUEVuOVAKT2hNaGVyZkExekVFVmkvSWZ3SURBUUFC")
```

Comment:
Alternatively, you could put the key and hash in constants and use them where necessary to avoid repeating the longs strings throughout the test